src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java

```diff
@@ -251,26 +251,32 @@ protected boolean shouldIncreaseIndent() {
     /**
      * Check the indentation for a set of lines.
      *
-     * @param lines              the set of lines to check
+     * @param astSet             the set of abstract syntax tree to check
      * @param indentLevel        the indentation level
      * @param firstLineMatches   whether or not the first line has to match
      * @param firstLine          first line of whole expression
+     * @param allowNesting       whether or not subtree nesting is allowed
```

Comment:
How is this related to the issue of logging violations by AST?

Reply:
Here are three types of columnNumbers
```
1. ast.columnNumber()
2. getLineStart()
3. expandedTabsCoumnNumber()
```
Earlier we were mapping lineNumbers with the ColumnNumber so there was no chance of confusion... as different conditions requires different columnNumbers...
https://github.com/checkstyle/checkstyle/blob/7b55584d331154daf37ab8f56b83be2951ef263f/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java#L395
```
        if (firstLineMatches && !allowNesting) {
            subtreeLines.addLineAndCol(firstLine,
                getLineStart(indentCheck.getLine(firstLine - 1)));
        }
```
e.g. here if above `if` statement satisfies than we were adding column no. given by `getLineStart()` else we were adding column no. given by `expandedTabsColumnNumber()`.

But Now with latest changes, we are mapping lines with AST, and an AST's column could be get by three types all may have different answers (see list above). 

So I have passed `allowNesting` parameter here to check the condition in this method and fetch correct column number inside it. 

It is been used here at line 274:
```
 final DetailAST startLineAst = astSet.firstLine();
            final int endLine = astSet.lastLine();
            int startCol = expandedTabsColumnNo(astSet.firstLine());

            final int realStartCol =
                getLineStart(indentCheck.getLine(startLineAst.getLineNo() - 1));

274-------> if (firstLineMatches && !allowNesting) {
                startCol = realStartCol;
            }

            if (realStartCol == startCol) {
                checkLineIndent(startLineAst, indentLevel,
                    firstLineMatches);
            }
```

Reply:
>  ast.columnNumber()

When we are checking indentation, we should never be using directly this. We should always be expanding this as tabs change the position of the column number based on the tab width.

> allowNesting 

When are these 2 values not equal? It seems to me when they are not equal there is an issue of some kind.

Reply:
> When we are checking indentation, we should never be using directly this

Yes, It is never used.

> When are these 2 values not equal? It seems to me when they are not equal there is an issue of some kind.

I have added an additional test-case for this, [here](https://github.com/checkstyle/checkstyle/pull/8038#discussion_r443647890)